File: 动态规划专题/509_斐波那契数.py

#动态规划   32ms
class Solution:
    def fib(self, n: int) -> int:
        sum = []
        sum.append(0)
        sum.append(1)
        if n > 1:
            for i in range(2, n + 1):
                sum.append(sum[i - 1] + sum[i - 2])
        return sum[n]
		
		
#递归  700ms
class Solution:
    def fib(self, n: int) -> int:
        if n < 2:
            return n
        return self.fib(n - 1) + self.fib(n - 2)

File: 动态规划专题/509_斐波那契数.cpp

//动态规划 0ms
class Solution {
public:
	int fib(int n) {
		vector<int> ret;
		ret.push_back(0);
		ret.push_back(1);
		if (n > 1)
		{
			for (int i = 2; i < n + 1; i++)
			{
				ret.push_back(ret[i - 1] + ret[i - 2]);
			}
		}
		return ret[n];
	}
};

//递归  20ms
class Solution {
public:
	int fib(int n) {
		if(n < 2)
        {
            return n;
        }
        return this->fib(n - 1) + this->fib(n - 2);
	}
};